### PR TITLE
Print gradients for SA-REKS states

### PR DESCRIPTION
**Pull request by inseonglee**
These gradients can be used as diabatic properties.

### CODE REVIEW

File: prog/dftb+/lib_reks/reksio.F90 [hunk 1]
```diff
@@ -180,18 +180,20 @@ subroutine printReksGradInfo(this, derivs)
...
-!          write(stdOut,*) ist, "st state (SA-REKS)"
-!          write(stdOut,'(3(f15.8))') this%SAgrad(:,:,ist)
-!          if (ist == this%nstates) then
-!            write(stdOut,"(A)") repeat("-", 50)
-!          else
-!            write(stdOut,'(3(f15.8))')
-!          end if
-!        end do
+        if (this%Plevel >= 2) then
+          write(stdOut,"(12X,A)") "AVG state"
```
Comment (bhourahine) [inline prog/dftb+/lib_reks/reksio.F90:184]: Any reason not to write AVG in full instead of as an acronym?

File: prog/dftb+/lib_reks/reksio.F90 [hunk 2]
```diff
@@ -180,18 +180,20 @@ subroutine printReksGradInfo(this, derivs)
...
+          write(stdOut,'(3(f15.8))') this%avgGrad(:,:)
+          write(stdOut,'(3(f15.8))')
+          do ist = 1, this%nstates
+            write(stdOut,*) ist, "st state (SA-REKS)"
```
Comment (bhourahine) [inline prog/dftb+/lib_reks/reksio.F90:188]: Might be better to make an array of term names, something like
character(3), parameter :: ordinals(6) = ['1st', '2nd', '3rd', '4th', '5th', '6th']
and
write(stdOut, *)ordinals(ist) // " state (SA-REKS)"

**Review by bhourahine — APPROVED** [review]
Seems ready to merge, but see some cosmetic comments.

File: prog/dftb+/lib_reks/reksio.F90 (hunk 1 above)
Comment (inseonglee) [reply prog/dftb+/lib_reks/reksio.F90]: Done.

File: prog/dftb+/lib_reks/reksio.F90 (hunk 2 above)
Comment (inseonglee) [reply prog/dftb+/lib_reks/reksio.F90]: Done.

**Pull request merged**